locallib.php

```diff
@@ -255,27 +255,36 @@ private function process_action_give_rating() {
                 $renderer->add_notification(get_string('no_choice_to_rate', ratingallocate_MOD_NAME));
             } else if ($status === self::DISTRIBUTION_STATUS_RATING_IN_PROGRESS) {
                 // Rating is possible...
-
+                $viewurl = new moodle_url('/mod/ratingallocate/view.php', ['id' => $this->coursemodule->id]);
+                $giveratingurl = new moodle_url('/mod/ratingallocate/view.php',
+                    ['id' => $this->coursemodule->id, 'action' => 'give_rating']);
                 // Suche das richtige Formular nach Strategie.
                 $strategyform = 'ratingallocate\\' . $this->ratingallocate->strategy . '\\mod_ratingallocate_view_form';
 
                 /* @var $mform moodleform */
                 $mform = new $strategyform($PAGE->url->out(), $this);
                 $mform->add_action_buttons();
-
                 if ( $mform->is_cancelled() ) {
                     // Return to view.
-                    redirect("$CFG->wwwroot/mod/ratingallocate/view.php?id=".$this->coursemodule->id);
-                    return "";
+                    redirect($viewurl);
```

Comment:
please also return an empty string :)

Reply:
thanks - I'll take a look - I can't remember why I changed to using a redirect in there...

Reply:
oh - are you asking me to return "" after the redirect? - I removed that because it doesn't do anything and won't get hit because the Moodle redirect function exits at the end - a call to "redirect" is basically the same as calling "exit" in the function so anything after that call will not be triggered.